[Fergus Mok] iP

FergusMok opened the pull request:
DukePro
> “Your mind is for having ideas, not holding them.” – David Allen (source)

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from ![here](https://nus-cs2103-ay2122s1.github.io/website/admin/ip-w4.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

FergusMok committed:
Add Level-1

FergusMok committed:
Add Level-2

FergusMok committed:
Add Level-3

FergusMok committed:
Add Level-4

FergusMok committed:
Add A-TextUITesting

FergusMok committed:
Add Level-5

FergusMok committed:
Add Level-6

FergusMok committed:
Add Level-7

FergusMok committed:
Add Level-8

FergusMok committed:
Merge branch 'branch-Level-7'

FergusMok committed:
Merge branch 'branch-Level-8'

FergusMok committed:
Add A-MoreOOP

FergusMok committed:
Add A-Packages

FergusMok committed:
Add A-JUnit

FergusMok committed:
Add A-Jar

FergusMok committed:
Add Level-9

FergusMok committed:
Add A-JavaDoc

FergusMok committed:
Add A-CodingStandard

FergusMok committed:
Merge branch 'branch-Level-9'

FergusMok committed:
Merge branch 'branch-A-CodingStandard'

FergusMok committed:
Merge branch 'branch-A-JavaDoc'

huyuxin0429 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+
+  private String date;
+
+  public Deadline(String description, String date) {
+    this.description = description;
+    this.date = date;
+    this.done = false;
+  }
```
There is some redundancy here, try this instead:
```suggestion
  public Deadline(String description, String date) {
    this(description, date, false)
  }
```

huyuxin0429 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+  public Deadline(String description, String date, boolean done) {
+    this.description = description;
+    this.date = date;
+    this.done = done;
```
```suggestion
    this.isDone = done;
```
Do you think using isDone will conform better to our styleguide?

huyuxin0429 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,40 @@
+  public Event(String description, String date, boolean done) {
+    this.description = description;
+    this.date = date;
+    this.done = done;
```
Similar issue of isDone

huyuxin0429 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,115 @@
+          }
+        case COMMAND_EVENT:
+          {
+            this.taskArray.handleAddEvent(commands);
+            break;
+          }
+        case COMMAND_DEADLINE:
+          {
+            this.taskArray.handleAddDeadline(commands);
+            break;
+          }
+        case COMMAND_TODO:
+          {
+            this.taskArray.handleAddToDo(commands);
+            break;
+          }
+        case COMMAND_DONE:
+          {
+            int taskIndex = Integer.parseInt(commands[1]);
+            this.taskArray.handleDone(taskIndex);
+            break;
+          }
+        case COMMAND_DELETE:
+          {
+            int taskIndex = Integer.parseInt(commands[1]);
+            this.taskArray.handleDelete(taskIndex);
+            break;
+          }
+        case COMMAND_SAVE:
+          {
+            this.taskArray.handleSave(this.storage);
+            break;
+          }
+        case COMMAND_FIND:
+          this.taskArray.handleFind(commands[1]);
+          break;
+        case COMMAND_RESET:
+          this.taskArray.handleReset();
+          break;
+        default:
```
Do you think switch and case should have the same indentation level to conform better with our styleguide?

huyuxin0429 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,32 @@
+public abstract class Task {
+
+  protected String description;
+  protected boolean done;
```
Do you think changing to isDone will better conform to our styleguide?
```suggestion
  protected boolean isDone;
```

huyuxin0429 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,202 @@
+      String command = commands[0];
+      boolean done = commands[1].equals("1");
+      String description = commands[2];
+      switch (command) {
+        case Duke.COMMAND_EVENT:
+          {
+            String date = commands[3];
+            newTaskArray.add(new Event(description, date, done));
+            break;
+          }
+        case Duke.COMMAND_DEADLINE:
+          {
+            String date = commands[3];
+            newTaskArray.add(new Deadline(description, date, done));
+            break;
+          }
+        case Duke.COMMAND_TODO:
+          {
+            newTaskArray.add(new Todo(description, done));
+            break;
+          }
+        default:
+          {
+            Printer.print(ERROR_UNKNOWN_FILE_COMMAND);
+          }
```
Should switch be on the same indentation as case to conform with styleguide?

huyuxin0429 commented on src/main/java/duke/DateFormatter.java:
```diff
@@ -0,0 +1,27 @@
+   * Formats date to use MMM d yyyy pattern
+   *
+   * @param date
+   * @return
```
I think filling in the @return tag will be informative.

huyuxin0429 reviewed (commented):
Good job there, some minor changes left

FergusMok committed:
Merge remote-tracking branch 'remotes/origin/add-gradle-support'

FergusMok committed:
Add A-Gradle

FergusMok committed:
Add .project

FergusMok committed:
Merge add-gradle-support

FergusMok committed:
Add A-CheckStyle

chunweii commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,111 @@
+
+public class Duke {
+
+    protected static final String COMMAND_TODO = "todo";
```
Long list of Command strings can be put into an enumeration.

chunweii commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,208 @@
+        }
+
+        Pattern pattern = Pattern.compile(description, Pattern.CASE_INSENSITIVE);
+        List<Task> tasksFound = taskArray
```
Consider abstracting this into another method to make this method shorter.

chunweii reviewed (commented):
All your JavaDocs should have their tag description filled up. Public classes can also have JavaDocs to describe what the purpose of the class.

FergusMok committed:
Add Level-10

FergusMok committed:
Merge branch 'branch-Level-10'

FergusMok committed:
Add code-quality enhancementts

FergusMok committed:
Add abstract DatedTask class

Event and Deadline constructors are similar, resulting in code
duplication.

Unifying all those duplicated code improves the code quality.
Using the DatedTask class instead of Task for Event and Deadline
functions will also be more accurate, resulting in less uncaught
problems in the future.

FergusMok committed:
Refactor Tasks into their own folder, and added Exceptions

Exceptions are currently caught and shown to the user as messages.

Exceptions should be thrown and handled, and should not be mixed together with correct responses.

FergusMok committed:
Add C-Help

FergusMok committed:
Add assertions

FergusMok committed:
Fix merge conflicts for C-Help

FergusMok committed:
Fix merge conflicts for A-CodeQuality

FergusMok committed:
Merge pull request #3 from FergusMok/branch-A-CodeQuality

Add A-CodeQuality

FergusMok committed:
Fix C-Help merge conflicts

FergusMok committed:
Merge pull request #5 from FergusMok/branch-C-Help

Add C-Help

FergusMok committed:
Fix merge conflicts for branch-A-Assertions

FergusMok committed:
Merge pull request #2 from FergusMok/branch-A-Assertions

Add A-Assertions

FergusMok committed:
Add a fix to DukeException thrown at setDuke

FergusMok committed:
Add more profile pictures, change font, add enums

FergusMok committed:
Add timer to bye command

FergusMok committed:
Add more exception-handling for tasks

FergusMok committed:
Merge branch 'branch-A-BetterGui'

FergusMok committed:
Fix checkstyle formatting

FergusMok committed:
Configure JUnit tests for gradle

FergusMok committed:
Set theme jekyll-theme-slate

FergusMok committed:
Add Ui.png

FergusMok committed:
Merge branch 'master' of github.com:FergusMok/ip

FergusMok committed:
Add README for user guide

FergusMok committed:
Change README

FergusMok committed:
Update build.gradle for release

FergusMok committed:
Rename Ui.png

FergusMok committed:
Rename Ui.png

FergusMok committed:
Merge branch 'master' of github.com:FergusMok/ip
